File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        st = set('aeiou')
        n = len(s)
        count = 0
        maxi = 0
        # first window
        for i in range(k):
            if s[i] in st:
                count += 1
        maxi = count

    
        # slide
        for i in range(k,n):
            if s[i] in st:
                count += 1
            if s[i-k] in st:
                count -= 1
            maxi = max(maxi, count)

        return maxi

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.cpp

class Solution {
public:
    int maxVowels(string s, int k) {
        int n = s.size();
        int ans = INT_MIN;
        int curr_window = 0;
        set<char> st = {'a','e','i','o','u'};

        // first window
        for(int i=0; i<k; i++){
            if(st.count(s[i])){
                curr_window++;
            }
        }
        ans =max(ans, curr_window);
        // slide
        for(int j=k; j<n; j++){
            if(st.count(s[j])){
                curr_window++;
            }
            if(st.count(s[j-k])){
                curr_window--;
            }
            ans = max(ans, curr_window);
        }
        return ans;
    }

};

